### DISCUSSION TITLE
Review of legal documents in PDF - suggestion needed.

### DISCUSSION THREAD

**Discussion by DoubleCortado** [explosion/spaCy#12361] [Help: Model Advice] [answered]
Hello everyone.
I would like to set up model to review documents generated by legal dept. Could you let me know which spacy tool would most suitable to solve below problems? If my description of issues is not sufficient please let me know- I will provide more details.

1. Recognizing tables in PDF and checking if rows with total values are correct summing all rows which are above. Table are coming from excel and then pasted to Word file which is converted to PDF. I have to make sure that always rows with totals are correct.

2. Date format consistency throughout report. Let’s say expected format date is December 31, 2022, I need to know if there are cases where date is in different format. I presume here I can use en_core_web_lg model and pull dates using DATE label.

3. Company name consistency - meaning if in one place company name is EVENTS S.A.  and somewhere else is EVENT SA  What would be the best way to perform such check?

I hope above can be solved using spaCy as I would be really excited to explore this universe.

Below is screen of tables mentioned in the first point with total rows in bold:
![Screenshot 2023-03-03 at 13 56 14](https://user-images.githubusercontent.com/72337625/222725933-810fa4dd-8432-492f-882c-b6f77c7dafe6.png)

**Comment by thomashacker** [comment, marked as answer]
Hey,
Thanks for your question. Here are my thoughts on your three use cases:

1. Unfortunately, spaCy isn't capable of reading in PDFs yet.
2. Sure, you can use the pretrained models to extract the `DATE` entities and check their formatting. But I think using regex to find all date formats might be more helpful.
3. Finding and extracting company names is a great use case for spaCy. You can use the pretrained NER models to extract company names and check if their names are different. You can also fine-tune models with your own data, or even train a model from scratch. You can find more information [here in our docs](https://spacy.io/usage/training).

I hope this was helpful!
